MathLib/LinAlg/PETSc/PETScVector.cpp

```diff
@@ -72,6 +76,13 @@ PETScVector::PETScVector(const PETScVector &existing_vec, const bool deep_copy)
     {
         VecCopy(existing_vec._v, _v);
     }
+
+    if (!_global_v)
```

Comment:
This is a copy ctor, so this condition always holds.
But is this code necessary here if we do `getGlobalVector()` anyway before any critical operation?